sdks/java/io/redis/src/test/java/org/apache/beam/sdk/io/redis/RedisIOTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.redis;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.ArrayList;
+
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.Rule;
+import org.junit.Test;
+
+import redis.clients.jedis.Jedis;
+import redis.embedded.RedisServer;
+
+/**
+ * Test on the Redis IO.
+ */
+public class RedisIOTest {
+
+  @Rule public TestPipeline pipeline = TestPipeline.create();
+
+  @Test
+  public void testGetRead() throws Exception {
+    try (EmbeddedRedis embeddedRedis = new EmbeddedRedis()) {
+      insertKV(embeddedRedis);
+
+      ArrayList<KV<String, String>> expected = new ArrayList<>();
+      for (int i = 0; i < 1000; i++) {
+        KV<String, String> kv = KV.of("Key " + i, "Value " + i);
+        expected.add(kv);
+      }
+
+      PCollection<KV<String, String>> readAll = pipeline.apply(
+          RedisIO.read()
+              .withConnectionConfiguration(createConnection(embeddedRedis)));
+
+      PAssert.that(readAll).containsInAnyOrder(expected);
+
+      pipeline.run();
+    }
+  }
+
+  @Test
+  public void testGetReadWithKeyPattern() throws Exception {
+    try (EmbeddedRedis embeddedRedis = new EmbeddedRedis()) {
+      insertKV(embeddedRedis);
+
+      ArrayList<KV<String, String>> expected = new ArrayList<>();
+      expected.add(KV.of("Key 10", "Value 10"));
+      for (int i = 100; i < 110; i++) {
+        expected.add(KV.of("Key " + i, "Value " + i));
+      }
+
+      PCollection<KV<String, String>> read = pipeline.apply(
+          RedisIO.read()
+            .withConnectionConfiguration(RedisConnectionConfiguration.create()
+              .withHost("localhost")
```

Comment:
This is the same as createConnection(embeddedRedis). I'd suggest to put it in a member variable.